### PR TITLE
3051 Doesn't select the searched entry from list in program enrolment form

### LINKED ISSUE #3051: Doesn't select the searched entry from list in program enrolment form
## How to Reproduce 🔨

Steps to reproduce the behaviour:
1. Get programs-setup datafile like PNG Demo
2. Have some patients with common starting letters like eg: Mimi Chan, Mimi Dan, etc
3. Create new patient and enrol it into a program like HIV Care & Treatment
4. Now in the enrolment form, in Mother field, put few characters in first name such that both previous names appear in search option, eg: put Mimi in first name and see both `Mimi Chan` and `Mimi Dan` appear in the list
5. Make sure to select the second one: `Mimi Dan`
6. See error: it automatically selects the first one : `Mimi Chan`

## Expected behaviour 🤔
It should select the one we select; not take the first entry by default! 

## Screenshot/Recording
https://recordit.co/S4MBEPHXbN

## Additional Context
Found while testing #1840 

## Your environment 🌱

- Version: 1.6.2
- mSupply version: 713-00-beta3
- Platform:
  - browser (extra points if you tell us which one)
  - server (windows)
- Database type: 
  - SQLite3

### PR DESCRIPTION
**Pull request by roxy-dao**
Fixes #3051

# 👩🏻‍💻 What does this PR do? 
Fix the patient check.

# 🧪 How has/should this change been tested? 
- [ ] Have programs set up
- [ ] Create two patients that have the same first name `e.g. Mimi` and a third with whatever name you want
- [ ] Enrol the third patient into `HIV Care and Treatment` program
- [ ] Go to the mother field and type `Mimi` or whatever name you have chosen
- [ ] Select the 2nd `Mimi` on the list
- [ ] Should load correctly

### CODE REVIEW

File: client/packages/programs/src/JsonForms/components/Search/SearchWithUserSource.tsx [hunk 1]
```diff
@@ -53,10 +53,10 @@ export const SearchWithUserSource = (
   const getOptionLabel = (data: PatientSchema) =>
     options?.optionString
       ? formatTemplateString(options?.optionString, data)
-      : `${data['code']} - ${data['firstName']} ${data['lastName']}`;
+      : `${data['code'] ?? ''} - ${data['firstName']} ${data['lastName']}`;
```
Comment (roxy-dao) [inline client/packages/programs/src/JsonForms/components/Search/SearchWithUserSource.tsx:56]: Code wasn't required.... so it showed up as undefined for me 👀  probably a better way to display this undefined value.... or make code required?

**Review by jmbrunskill — APPROVED** [review]
Haven't tested but looks like a valid code change :)

Ok to leave testing to the testers?

File: client/packages/programs/src/JsonForms/components/Search/SearchWithUserSource.tsx (hunk 1 above)
Comment (clemens-msupply) [reply client/packages/programs/src/JsonForms/components/Search/SearchWithUserSource.tsx:56]: If there is no code we shouldn't show the extra " - ", right? e.g.

```
`${data['code'] ?? `${data['code']} - `}${data['firstName']} ${data['lastName']}`;
```

Comment (roxy-dao) [reply client/packages/programs/src/JsonForms/components/Search/SearchWithUserSource.tsx]: Good idea~

**Comment by roxy-dao** [comment]
> Haven't tested but looks like a valid code change :)
> 
> Ok to leave testing to the testers?

Yup~ Happy to leave it to testers if you think code looks good (:

**Pull request merged**